Source/Core/DolphinQt/Debugger/NetworkWidget.h

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <QDockWidget>
+
+#include "Common/CommonTypes.h"
+
+class QCheckBox;
+class QCloseEvent;
+class QGroupBox;
+class QShowEvent;
+class QTableWidget;
+class QTableWidgetItem;
+
+class NetworkWidget : public QDockWidget
+{
+  Q_OBJECT
+public:
+  explicit NetworkWidget(QWidget* parent = nullptr);
+  ~NetworkWidget();
+
+protected:
+  void closeEvent(QCloseEvent*) override;
+  void showEvent(QShowEvent* event) override;
+
+private:
+  void CreateWidgets();
+  void ConnectWidgets();
+
+  void Update();
+
+  QGroupBox* CreateSocketTableGroup();
+  QGroupBox* CreateSSLContextGroup();
+  QGroupBox* CreateSSLOptionsGroup();
+
+  QTableWidgetItem* GetSocketDomain(s32 host_fd) const;
```

Comment:
As far as I can tell, these don't have to be member functions and exposed in the header.